portfolio/src/main/java/com/google/sps/data/Comment.java

```diff
@@ -15,25 +15,33 @@
 package com.google.sps.data;
 
 import com.google.auto.value.AutoValue;
+import java.util.Optional;
 
 /** Stores data for comments written in portfolio. */
 @AutoValue 
 public abstract class Comment {
   public abstract long id();
   public abstract String name();
   public abstract String email();
+  public abstract Optional<String> imageUrl();
   public abstract String commentText();
   public abstract long timeStamp();
 
-  /** 
-   * Creates a Comment.
-   * @param id Id of object stored in database.
-   * @param name Name of author.
-   * @param email Email of user comment was written by.
-   * @param commentText Text author wrote.
-   * @param timeStamp Time when author submitted comment.
-   */
-  public static Comment create(long id, String name, String email, String commentText, long timeStamp) {
-    return new AutoValue_Comment(id, name, email, commentText, timeStamp);
+  /** Returns a Builder for a Comment. */
+  public static Builder builder(long id, String name, String email, String commentText, long timeStamp) {
```

Comment:
I wondering if we should just follow convention and have
```
public static Builder builder() {
  return new AutoValue_Comment.Builder();
}
```
Then let the caller do .set(..).set(...)...
It's a bit weird to have a builder with parameters, doesn't it defy the reason for a builder in the first place?
The builder design pattern is used where having so many arguments is not palatable. 
Having a builder() return and we then chain the initializations is the desire with this pattern.

https://www.geeksforgeeks.org/builder-design-pattern/
The pattern seeks to separate object initialization from representation.

Reply:
fixed